com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/activiti/ActivitiFacade.java

```diff
@@ -67,16 +70,22 @@ public ProcessInstance startProcess(String userId, String processDefinitionKey,
 
     public HistoricProcessInstance getHistoricProcessInstanceBySpaceId(String processDefinitionKey, String spaceId,
         String processInstanceId) {
-        HistoricProcessInstanceQuery query = engine.getHistoryService().createHistoricProcessInstanceQuery().processDefinitionKey(
-            processDefinitionKey).variableValueEquals(Constants.VARIABLE_NAME_SPACE_ID, spaceId).excludeSubprocesses(
-                true).processInstanceId(processInstanceId);
+        HistoricProcessInstanceQuery query = engine.getHistoryService()
+            .createHistoricProcessInstanceQuery()
+            .processDefinitionKey(processDefinitionKey)
+            .variableValueEquals(Constants.VARIABLE_NAME_SPACE_ID, spaceId)
+            .excludeSubprocesses(true)
+            .processInstanceId(processInstanceId);
 
         return query.singleResult();
     }
 
-    public HistoricProcessInstance getHistoricProcessInstanceByProcessInstanceId(String processInstanceId) {
-        return engine.getHistoryService().createHistoricProcessInstanceQuery().processInstanceId(processInstanceId).singleResult();
+    public List<HistoricProcessInstance> getHistoricProcessInstancesFinishedAndStartedBefore(Date expirationTime) {
+        return engine.getHistoryService().createHistoricProcessInstanceQuery().finished().startedBefore(expirationTime).list();
+    }
 
+    public void deleteHistoricProcessInstance(String processInstanceId) {
+        engine.getHistoryService().deleteHistoricProcessInstance(processInstanceId);
```

Comment:
Would this cascade delete also the historicProcessVariables?

Reply:
Yes, it deletes them as well